R/teal_data-datanames.R

```diff
@@ -43,10 +43,25 @@ setGeneric("datanames<-", function(x, value) standardGeneric("datanames<-"))
 setMethod("datanames<-", signature = c("teal_data", "character"), definition = function(x, value) {
   checkmate::assert_subset(value, names(x@env))
   x@datanames <- value
+  x@datanames <- sort_datanames(x@datanames, x@join_keys)
```

Comment:
also needed in getter function as one can set `join_keys` after setting `datanames`

Reply:
If we stick with `join_keys<-` changing the `x@datanames` field then it's ok and getter does not need to be changeed